server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -194,7 +194,7 @@ public ActionRequestValidationException validate() {
         validationException = DocWriteRequest.validateSeqNoBasedCASParams(this, validationException);
 
         if (id != null && id.getBytes(StandardCharsets.UTF_8).length > 512) {
-            validationException = addValidationError("id is too long, must be no longer than 512 bytes but was: " +
```

Comment:
We prefer to include the id in square brackets. Can you change the message to "id [" + id + "] is too long ...".

Reply:
Make sense to me. Glad to change it.